topostats/run_topostats.py

```diff
@@ -146,61 +185,77 @@ def process_scan(
     filtered_image.filter_image()
 
     # Find Grains :
-    #
-    # The Grains class also has a convenience method that runs the instantiated class in full.
-    grains = Grains(
-        image=filtered_image.images["zero_averaged_background"],
-        filename=filtered_image.filename,
-        pixel_to_nm_scaling=filtered_image.pixel_to_nm_scaling,
-        threshold_method=threshold_method,
-        gaussian_size=gaussian_size,
-        gaussian_mode=gaussian_mode,
-        threshold_multiplier=threshold_multiplier,
-        background=background,
-    )
-    grains.find_grains()
+    try:
+        LOGGER.info(f"[{filtered_image.filename}] : Grain Finding")
+        grains = Grains(
+            image=filtered_image.images["zero_averaged_background"],
+            filename=filtered_image.filename,
+            pixel_to_nm_scaling=filtered_image.pixel_to_nm_scaling,
+            threshold_method=threshold_method,
+            gaussian_size=gaussian_size,
+            gaussian_mode=gaussian_mode,
+            threshold_multiplier=threshold_multiplier,
+            background=background,
+        )
+        grains.find_grains()
+    except IndexError as index_exception:
+        LOGGER.info(
+            f"[{filtered_image.filename}] : No grains were detected, skipping Grain Statistics and DNA Tracing."
+        )
 
-    # Grain Statistics :
-    #
-    # Ditto for the GrainStats class.
-    grainstats = GrainStats(
-        data=grains.images["gaussian_filtered"],
-        labelled_data=grains.images["labelled_regions"],
-        pixel_to_nanometre_scaling=filtered_image.pixel_to_nm_scaling,
-        img_name=filtered_image.filename,
-        output_dir=output_dir / filtered_image.filename,
-    )
-    grain_statistics = grainstats.calculate_stats()
+    if len(grains.region_properties) > 0:
+        # Grain Statistics :
+        try:
```

Comment:
Why might this fail if the condition at 206 is passed?

Reply:
One instance I've found so far is a `ValueError` is raised as a consequence of some very small grains being detected.  At the moment these are captured with a very broad `Exception` but I expect the `ValueError` I have found to not occur once we remove "noise" based on a minimum grain size.

But I suspect there is probably a lot of other scenarios that might cause errors and would be keen to introduce regression tests for more sample types, `silicon`, `water`, `proteins` and so forth to be able to get more specific exceptions defined and handled.

For now though I've left the broad `try: .... except:` that returns an empty data frame of statistics for that image so that when processing multiple images it "fails gracefully" if one causes a problem.